### DISCUSSION TITLE
is there any Customizable Sniff Properties for classes/traits/interfaces ?

### DISCUSSION THREAD

**Discussion by mmmsafry** [squizlabs/PHP_CodeSniffer#3423] [Q&A]
is there any Customizable Sniff Properties for classes/traits/interfaces to do the followings;

- not allow a blank line before the first content 
- One blank line after the last content 

I am expecting a rule for something like this.
```
Class Example()
{
  /** Content */

}
```

**Comment by jrfnl** [comment]
You may want to have a look at the `Squiz.WhiteSpace.FunctionSpacing` sniff.
Not completely what you want, but as most classes will have functions at the end, the "One line after last" can be enforced and if the class content starts with a function, the "No blank line before first" as well.

See: https://github.com/squizlabs/PHP_CodeSniffer/wiki/Customisable-Sniff-Properties#squizwhitespacefunctionspacing

P.S.: Why the heck did you ping me in this topic ? You think I'm your personal helpdesk ? That's just rude.

**Reply by mmmsafry** [reply]
@jrfnl sorry about the personal ping, this is my first discussion question, I just recently saw your video about phpcs, that was great and helpful. again sorry for personal ping. Its removed.
